datacube/api/query.py

```diff
@@ -153,7 +158,8 @@ def __init__(
             assert self.geopolygon is None, (
                 "'like' with other spatial bounding parameters is not supported"
             )
-            self.geopolygon = getattr(like, "extent", self.geopolygon)
+            like = _normalise_geobox(like)
+            self.geopolygon = like.extent
 
             if "time" not in self.search:
                 time_coord = like.coords.get("time")  # type: ignore[union-attr]
```

Comment:
I think you need to change this `union-attr` with `attr-defined` to appease mypy.